modules/gapi/test/common/gapi_tests_common.hpp

```diff
@@ -580,6 +696,27 @@ struct WrappableVector
     }
 };
 
+template<typename T, typename Elem, int cn>
+struct WrappableVec
+{
+    compare_vec_f<Elem, cn> to_compare_f()
+    {
+        T t = *static_cast<T* const>(this);
```

Comment:
Why not `T t = static_cast<T>(*this);` ?

Reply:
We should cast pointer as otherwise it doesn't work. And `const` is just for safety

Actually, this's not good o cast base class object to derived one.. I think I'll raise the discussion about it